src/audio/smart_amp_test.c

```diff
@@ -185,17 +185,6 @@ static int smart_amp_ctrl_set_bin_data(struct comp_dev *dev,
 
 	assert(sad);
 
-	if (dev->state != COMP_STATE_READY) {
```

Comment:
@RDharageswari do you think we need to check for the case when state is < READY? Can we do this during init? We want to wait until the initialize params are set before we modify them isnt it?

Reply:
Done